flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/BufferResponseDecoderDelegate.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.runtime.io.network.NetworkClientHandler;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.channel.ChannelHandlerContext;
+
+import static org.apache.flink.runtime.io.network.netty.NettyMessage.BufferResponse;
+
+/**
+ * The parser for {@link org.apache.flink.runtime.io.network.netty.NettyMessage.BufferResponse}.
+ */
+public class BufferResponseDecoderDelegate implements NettyMessageDecoderDelegate {
+
+	/** The network client handler of current channel. */
+	private final NetworkClientHandler networkClientHandler;
+
+	/** The Flink Buffer allocator. */
+	private final NetworkBufferAllocator allocator;
+
+	/** The cumulation buffer of message header. */
+	private ByteBuf messageHeaderBuffer;
+
+	/**
+	 * The current BufferResponse message that are process the buffer part.
+	 * If it is null, we are still processing the message header part, otherwise
+	 * we are processing the buffer part.
+	 */
+	private BufferResponse currentResponse;
+
+	/** How much bytes have been received or discarded for the buffer part. */
+	private int decodedBytesOfBuffer;
+
+	public BufferResponseDecoderDelegate(NetworkClientHandler networkClientHandler) {
+		this.networkClientHandler = networkClientHandler;
```

Comment:
Do not need this class-level var based on the [comment](https://github.com/apache/flink/pull/7368/files#r381770789)

Reply:
Has modified accordingly.